lib/srv6.c

```diff
@@ -116,3 +121,46 @@ const char *seg6local_context2str(char *str, size_t size,
 		return str;
 	}
 }
+
+struct srv6_locator *srv6_locator_alloc(const char *name)
+{
+	struct srv6_locator *locator = NULL;
+
+	locator = XCALLOC(MTYPE_SRV6_LOCATOR, sizeof(struct srv6_locator));
+	strlcpy(locator->name, name, sizeof(locator->name));
+	locator->functions = list_new();
+	QOBJ_REG(locator, srv6_locator);
+	return locator;
+}
+
+void srv6_locator_free(struct srv6_locator *locator)
+{
+	list_delete(&locator->functions);
+	XFREE(MTYPE_SRV6_LOCATOR, locator);
+}
+
+struct srv6_function *srv6_function_alloc(const struct prefix_ipv6 *prefix)
+{
+	struct srv6_function *function = NULL;
+
+	function = XCALLOC(MTYPE_SRV6_FUNCTION, sizeof(struct srv6_function));
+	function->prefix = *prefix;
+	return function;
+}
+
+void srv6_function_init(struct srv6_function *function,
+			const char *locator_name,
+			const struct prefix_ipv6 *prefix)
+{
+	memset(function, 0, sizeof(*function));
```

Comment:
For some reason I thought we were using xmalloc or some such -- just checking to make certain this memset is really needed here.

Reply:
Yes, your saying is really right. :) 

In this case, we don't need to clear this var.
The caller of this function (srv6_function_init) should clear the this memory-area. (with xmalloc, etc..).

So, I'll delete this memset's line.
Thank you for your review.

Reply:
I fixed it.